### PR TITLE
feat: Update Wireguard example to new layout

### PR DESCRIPTION
**Pull request by bryantbiggs**
# Description

- Update Wireguard example to new layout; this is open for discussion - see motivation below
	- Since this is no longer a module, we do not have to follow the norms and conventions of a module (i.e. - `variables.tf`, `outputs.tf`, etc.). We are free to create the file naming and directory structure that makes it easier for users to discover and reach the information they are seeking
	- The sample resources were converted to use the Kubernetes provider to reduce our reliance on the kubectl provider which appears to have become defunct; this also removes the boilerplate of yet one more provider in the pattern. These resources are defined in `sample.tf` to make it clear this is not required by the pattern and can be removed/ignored
	- The EKS and Cilium resources were moved to `eks.tf` - this is the heart of where users should look when viewing code and this is highlighted in the docs new section `Areas of Interest` (this phrasing is open to change - I'm not super thrilled with it currently but the point still stands - "Look here for whats notable in this pattern")

### Motivation and Context

- The goal of this proposed change is to de-emphasize the boilerplate and supporting resources required for a stable pattern that is deployable, while emphasizing and highlighting the components that are notable and of interest for a pattern
	- Does this do that? Are there ways to improve on this? 

### How was this change tested?

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

### Additional Notes

- I have deployed and validated the example and updated the docs to reflect additional tests provided by the Cilium project

### CODE REVIEW

File: patterns/wireguard-with-cilium/README.md [hunk 1]
```diff
@@ -1,27 +1,56 @@
...
 
-This pattern demonstrates Cilium configured in CNI chaining mode with VPC CNI and with Wireguard transparent encryption enabled on an Amazon EKS cluster.
+This pattern demonstrates Cilium configured in CNI chaining mode with the VPC CNI and with Wireguard transparent encryption enabled on an Amazon EKS cluster.
 
-- [Cilium CNI Chaining Documentation](https://docs.cilium.io/en/v1.12/gettingstarted/cni-chaining-aws-cni/)
-- [Cilium Wireguard Encryption Documentation](https://docs.cilium.io/en/v1.12/gettingstarted/encryption-wireguard/)
+- [Cilium CNI Chaining Documentation](https://docs.cilium.io/en/stable/installation/cni-chaining-aws-cni/)
+- [Cilium Wireguard Encryption Documentation](https://docs.cilium.io/en/stable/security/network/encryption-wireguard/)
+
+## Areas of Interest
```
Comment (askulkarni2) [inline patterns/wireguard-with-cilium/README.md:8]: Key focal points - just a thought.

**Review by askulkarni2 — APPROVED** [review]
🚀

File: patterns/wireguard-with-cilium/README.md (hunk 1 above)
Comment (bryantbiggs) [reply patterns/wireguard-with-cilium/README.md:8]: `Focal Points` sounds good - short and sweet, easy to understand the intent

**Review by vara-bonthu — APPROVED** [review]

File: patterns/wireguard-with-cilium/README.md (hunk 1 above)
Comment (bryantbiggs) [reply patterns/wireguard-with-cilium/README.md]: updated in https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1761/commits/143ea9da75088d0247ed997fb17e3b30cf84c08a

**Review by askulkarni2 — APPROVED** [review]
LGTM!

**Pull request merged**